migen/build/platforms/nexsys3.py

```diff
@@ -0,0 +1,181 @@
+from migen.build.generic_platform import *
+from migen.build.xilinx import XilinxPlatform
+from migen.build.xilinx.programmer import Adept
+
+
+
+_io = [
+    ("clk100", 0, Pins("V12"), IOStandard("LVCMOS33")),
+
+    ("serial", 0,
+        Subsignal("tx", Pins("N17")),
+        Subsignal("rx", Pins("N18")),
+
+        IOStandard("LVCMOS33"),
+        Misc("SLEW=FAST")
+    ),
+
+    ("joystick", 0,
+        Subsignal("L", Pins("C4")),
+        Subsignal("R", Pins("D9")),
+        Subsignal("U", Pins("A8")),
+        Subsignal("D", Pins("C9")),
+        Subsignal("S", Pins("B8")),
+
+        Misc("PULLUP"),
+        IOStandard("LVCMOS33")
+    ),
+
+    ("dipswitch", 0, Pins("T10"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 1, Pins("T9"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 2, Pins("V9"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 3, Pins("M8"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 4, Pins("N8"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 5, Pins("U8"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 6, Pins("V8"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+    ("dipswitch", 7, Pins("T5"), IOStandard("LVCMOS33"), Misc("PULLUP")),
+
+    ("user_led", 0, Pins("U16"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 1, Pins("V16"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 2, Pins("U15"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 3, Pins("V15"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 4, Pins("M11"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 5, Pins("N11"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 6, Pins("R11"), IOStandard("LVCMOS33"), Drive(8)),
+    ("user_led", 7, Pins("T11"), IOStandard("LVCMOS33"), Drive(8)),
+
+
+    ("sevenseg", 0,
+        Subsignal("segment7", Pins("T17")),  # A
+        Subsignal("segment6", Pins("T18")),  # B
+        Subsignal("segment5", Pins("U17")),  # C
+        Subsignal("segment4", Pins("U18")),  # D
+        Subsignal("segment3", Pins("M14")),  # E
+        Subsignal("segment2", Pins("N14")),  # F
+        Subsignal("segment1", Pins("L14")),  # G
+        Subsignal("segment0", Pins("M13")),  # Dot
+        Subsignal("enable0", Pins("P17")),   # EN0
+        Subsignal("enable1", Pins("P18")),   # EN1
+        Subsignal("enable2", Pins("N15")),   # EN2
+        Subsignal("enable3", Pins("N16")),    # EN3
+
+        IOStandard("LVCMOS33")
+    ),
+
+
+    ("vga_out", 0,
+        Subsignal("hsync", Pins("N6")),
+        Subsignal("vsync", Pins("P7")),
+        Subsignal("r", Pins("U7 V7 N7")),
+        Subsignal("g", Pins("P8 T6 V6")),
+        Subsignal("b", Pins("R7 T7")),
+
+        IOStandard("LVCMOS33"),
+        Misc("SLEW=FAST")
+    ),
+
+    ("ethernet", 0,
+        Subsignal("PhyRstn", Pins("P3")),
```

Comment:
Please use names consistent with other boards (see e.g. M1 which also has MII), and, for subsignals, using lowercase and underscores (as they will end up being Python attributes).